### PR TITLE
Create ssh_device.erl

### PR DESCRIPTION
**Pull request by shijiang1130**
given another example by ssh library

### CODE REVIEW

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

File: lib/ssh/examples/ssh_device.erl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ssh:start(),
+    case ssh:connect(H, P,[{user, U}, {password, Pass},{silently_accept_hosts, true},{quiet_mode,true}]) of
+    	{ok, Conn} ->
+    		{ok, ChannelId} = ssh_connection:session_channel(Conn,infinity),
+    		ssh_connection:exec(Conn, ChannelId, Cmd,infinity),
+    		Init_rep = <<>>,
+    		wait_for_response(Conn,H,Init_rep),
+    		ssh:close(Conn);
+    	{error,nxdomain}->
+    	{ok,Try1}=ssh:connect(vnet:q_ip(H), P,[{user, U}, {password, Pass},{silently_accept_hosts, true},{quiet_mode,true}]),
```
Comment (tuncer) [inline lib/ssh/examples/ssh_device.erl]: Maybe this needs style fixes:
* missing spaces around `=`
* overlong line

**Comment by shijiang1130** [comment]
Thanks tuncer.  updated per your comments.

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

File: lib/ssh/examples/ssh_device.erl [hunk 2]
```diff
@@ -0,0 +1,71 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2005-2010. All Rights Reserved.
```
Comment (HansN) [inline lib/ssh/examples/ssh_device.erl]: The copyright notice is dated "-2010", it should be "-2015"

File: lib/ssh/examples/ssh_device.erl [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	of
+      {ok, Conn} ->
+	  {ok, ChannelId} = ssh_connection:session_channel(Conn,
+							   infinity),
+	  ssh_connection:exec(Conn, ChannelId, Cmd, infinity),
+	  Init_rep = <<>>,
+	  wait_for_response(Conn, H, Init_rep),
+	  ssh:close(Conn);
+      {error, nxdomain} ->
+	  {ok, Try1} = ssh:connect(vnet:q_ip(H), P,
```
Comment (HansN) [inline lib/ssh/examples/ssh_device.erl]: Where is the module vnet? I can't find it in Erlang/OTP

File: lib/ssh/examples/ssh_device.erl [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+%%
+
+%%
+
+-module(ssh_device).
+
+%% api
+-export([ssh_device/5]).
+
+ssh_device(H, P, U, Pass, Cmd) ->
```
Comment (HansN) [inline lib/ssh/examples/ssh_device.erl]: The variable names are not descriptive. Change H to Host, P to Port etc

File: lib/ssh/examples/ssh_device.erl [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+%%--------------------------------------------------------------------
+%%% Internal application API
+%%--------------------------------------------------------------------
+wait_for_response(Conn, H, Acc) ->
+    receive
+      {ssh_cm, Conn, Msg} ->
+	  case Msg of
+	    {closed, _ChannelId} ->
+		%file:write_file(string:concat("log\\",H), Acc, [append]),
```
Comment (HansN) [inline lib/ssh/examples/ssh_device.erl]: Please remove or un-comment.

File: lib/ssh/examples/ssh_device.erl [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		io:format([">>>>>>>>>>>>>>>>>>> ", H,
+			   " <<<<<<<<<<<<<<<<<\r\n", Acc, "\r\n"]);
+	    {data, _, _, A} ->
+		Acc2 = <<Acc/binary, A/binary>>,
+		wait_for_response(Conn, H, Acc2);
+	    _ ->
+		%io:format("~p", [Msg]),
```
Comment (HansN) [inline lib/ssh/examples/ssh_device.erl]: Please remove or un-comment.

**Comment by HansN** [comment]
I have reviewed your patch and have some comments and questions in addition to the in-line comments under "Files changed":

1) What does the module do?  A couple of comment lines in the beginning would help.
2) Add the module to the Makefile
3) There is code repeated in Lines 36-39 and lines 47-50 that would make a good separate function.
4) For code relaying on a remote peer behaving well, there is always a good habit to add a timeout if the peer doesn't send the closing message.  So I would like to see an after clause in the receive.
5) And finally, I would prefer the wait_for_responses/3 to respond with "{ok,Acc}|{error,timeout}"

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by zhird** [comment]
@shijiang1130, please squash this to one commit and write a good commit message for it (https://github.com/erlang/otp/wiki/Writing-good-commit-messages). As soon as that's done I'll merge it to maint :)

**Comment by shijiang1130** [comment]
- Add a ssh client example for quick using the SSH module.

**Comment by OTP-Maintainer** [comment]
The summary line of the commit message is too long and/or ends with a "."
Make sure the whole message follows the guidelines here: https://github.com/erlang/otp/wiki/Writing-good-commit-messages.

Bad message: Add a ssh client example for quick using the SSH module.


----------------

I am a script, I am not human 

----------------

**Comment by zhird** [comment]
@shijiang1130, how is it going with the commit squashing?

**Comment by shijiang1130** [comment]
@zhird , it's ok. thanks

**Comment by zhird** [comment]
@shijiang1130, I'm confused now; not sure why this was closed? We approved the change, I just wanted it squashed from 6 commits to 1 before merging it to master. I can fix that though if you're not sure about how to do it :) I'll reopen this since it still isn't merged.

**Comment by shijiang1130** [comment]
@zhird , Oh, i though you can merge the code even marked with closed, then i closed it. ^_^

**Comment by OTP-Maintainer** [comment]
The summary line of the commit message is too long and/or ends with a "."
Make sure the whole message follows the guidelines here: https://github.com/erlang/otp/wiki/Writing-good-commit-messages.

Bad message: Add a ssh client example for quick using the SSH module.


----------------

I am a script, I am not human 

----------------

**Comment by zhird** [comment]
Merged to maint.

**Pull request closed without merging**